io/src/main/scala/sbt/internal/io/EventMonitor.scala

```diff
@@ -127,12 +127,12 @@ private[sbt] object EventMonitor {
         recentEvents = recentEvents.filterNot(_._2.isOverdue)
         getFilesForKey(s.service.poll(delay)).foreach(maybeTrigger)
       }
-      def getFilesForKey(key: WatchKey): collection.Seq[Path] = key match {
-        case null => Nil
+      def getFilesForKey(key: WatchKey): Vector[Path] = key match {
```

Comment:
As a rule, I try to leave apis as generic as possible, so I'd probably use `IndexedSeq` over `Vector`, but I don't have strong feelings about this. This isn't a public api anyway.

Reply:
Eugene argued for Vector over Iterable, and I didn't have strong feelings about it either.

Reply:
My general preference is concrete datatypes instead of mixing and matching - https://www.scala-sbt.org/1.x/docs/Coding-Guideline.html#Use+specific+datatypes+%28%2C+%2C+or+%29%2C+rather+than
We might still opt to keep things abstract for performance reasons, but I think that should an exception not the rule.